docs: add page for community components

jo-chemla opened the pull request:
Following the issue https://github.com/pmndrs/react-three-fiber/issues/3377 I opened earlier today, here is a PR that adds a page to the `Getting Started` section listing external/community r3f components that have not been merged to drei or part of the pmndrs github org. Preview is [here](https://jo-chemla.github.io/react-three-fiber/getting-started/community-r3f-components)! 

I've added all the components I know of, feel free to add more before merging!

Not docs does not seem to take the `screenshot_url` param [visible here](https://github.com/pmndrs/docs/blob/5e994740f62bc7234b1cb1cec364052dfdd86c15/src/components/mdx/Codesandbox/Codesandbox.tsx#L17) of the `<Codesandbox />` component.

abernier commented:
as discussed on X, maybe a bit redundant with Discord's Showcase channel i think

abernier commented:
or we could write a `<DiscordShowcases staffPicked>` component, that displays all "staff-picked" labeled showcases, but again it would be redundant with such view that Discord already implement https://discord.com/channels/740090768164651008/1047453248044412968

=> at least (without implementing anything into docs.) we can introduce that `staff-picked` tag

jo-chemla commented:
Thanks for the feedback on X and here! Here is my take regarding the points you mentioned: 

I've been a long-time user of r3f/drei, but never saw a community collection of projects that did gather large threejs projects that expose react/r3f components. I'm thinking about large-scale projects rather than small set of utility reusable components, like other kind of renderers - 3d-tiles renderers from gkjohnson or nytimes r&d, luma labs gaussian splat renderer, looking glass etc. It would make sense to integrate it within the r3f doc for both discoverability - for those not on discord, or who rely a lot on exploring doc + github - and showcase, which could also in turn facilitate other teams/devs to add value to the fiber ecosystem. 

Regarding maintenance/curation of the selection of projects to add to this page, this should probably be at the discretion of the pmndrs team which hosts the doc page, with community contributions for adding new projects or removing outdated ones suggested via PRs the same way anyone could contribute more codesandboxes examples to the example docs page. 

I'm in favor of dissociating pages for 
 - showcase + examples, the existing one - which often embed small components within the sandboxes themselves, but could also be stored on outside repos - great for demo/learning/reuse of utilities
 - from full-fledged projects like the 3D-tiles renderers, CustomShaderMaterial or the luma and looking-glass renderers, which are more complex, dedicated to a scoped use that will probably never be integrated into drei or a pmndrs repo. 

Looking forward to hearing your thoughts!

jo-chemla commented:
Hi there, is there any feedback from others - either from the pmndrs team or outside - regarding this PR? I've integrated the feedback from @abernier to merge the `community components` into a section of the `showcase` page, but I'm open to extract it back to its own page if deemed useful. 

Looking forward to hearing your feedback!

jo-chemla commented:
Coming back on this after the new year, I think it made more sense to explode the existing, doc `Examples` page and the one I added via this PR, `Community Components`, which references heavier r3f projects I split among `Data sources, Renderers & frameworks, Materials and Utilities`. 

Any thoughts on this from the pmndrs team?

CodyJasonBennett commented:
This got missed. @drcmda, what do you think of this addition? I know you had full authorship of examples prior.

jo-chemla commented:
Current version of the page is accessible [here](https://jo-chemla.github.io/react-three-fiber/getting-started/community-r3f-components) - just added the incredible takram clouds and atmosphere repos if you want a quick look. Don't hesitate if you need any edits. 

Note I stumbled on the pmndrs/docs repo on something I think is a minor bug, which prevents the user from setting eg `screenshot_url` to codesandboxes, due to the fact user-provided props and data fetched from CSB is destructured the wrong way around. PR is [here](https://github.com/pmndrs/docs/pull/375)

jo-chemla commented:
Hi again, just did the final touch-ups following the merge of my fix for pmndrs/docs, so I could add original CodeSandboxes from the official repos of each project, with a user-controlled screenshot-urls for those that were missing. 

Everything is good to go from my point-of-view!

CodyJasonBennett commented on docs/getting-started/community-r3f-components.mdx:
```diff
@@ -0,0 +1,92 @@
+#### Materials
+
+ - [FarazzShaikh/CustomShaderMaterial](https://github.com/FarazzShaikh/THREE-CustomShaderMaterial)
+ - [spite/THREE.MeshLine](https://github.com/spite/THREE.MeshLine)
```
This is now https://github.com/pmndrs/meshline. Only Troika among these is maintained, though.

jo-chemla replied:
Thanks for the feedback, just edited meshline repo and CSB. Do you want me to flag the projects that seem not maintained or leave them the way they are now?
